src/features/soft_navigations/aggregate/index.js

```diff
@@ -0,0 +1,254 @@
+import { getConfigurationValue } from '../../../common/config/config'
+import { handle } from '../../../common/event-emitter/handle'
+import { registerHandler } from '../../../common/event-emitter/register-handler'
+import { HarvestScheduler } from '../../../common/harvest/harvest-scheduler'
+import { single } from '../../../common/util/invoke'
+import { timeToFirstByte } from '../../../common/vitals/time-to-first-byte'
+import { FEATURE_NAMES } from '../../../loaders/features/features'
+import { SUPPORTABILITY_METRIC_CHANNEL } from '../../metrics/constants'
+import { AggregateBase } from '../../utils/aggregate-base'
+import { API_TRIGGER_NAME, FEATURE_NAME, INTERACTION_STATUS } from '../constants'
+import { AjaxNode } from './ajax-node'
+import { InitialPageLoadInteraction } from './initial-page-load-interaction'
+import { Interaction } from './interaction'
+
+export class Aggregate extends AggregateBase {
+  static featureName = FEATURE_NAME
+  constructor (agentIdentifier, aggregator, { domObserver }) {
+    super(agentIdentifier, aggregator, FEATURE_NAME)
+
+    const harvestTimeSeconds = getConfigurationValue(agentIdentifier, 'spa.harvestTimeSeconds') || 10
+    this.interactionsToHarvest = []
+    this.interactionsAwaitingRetry = []
+    this.domObserver = domObserver
+
+    this.scheduler = new HarvestScheduler('events', {
+      onFinished: this.onHarvestFinished.bind(this),
+      retryDelay: harvestTimeSeconds,
+      onUnload: () => this.interactionInProgress?.done() // return any held ajax or jserr events so they can be sent with EoL harvest
+    }, { agentIdentifier, ee: this.ee })
+    this.scheduler.harvest.on('events', this.onHarvestStarted.bind(this))
+
+    this.initialPageLoadInteraction = new InitialPageLoadInteraction(agentIdentifier)
+    timeToFirstByte.subscribe(({ entries }) => {
+      const loadEventTime = entries[0].loadEventEnd
+      this.initialPageLoadInteraction.forceSave = true
+      this.initialPageLoadInteraction.done(loadEventTime)
+      this.interactionsToHarvest.push(this.initialPageLoadInteraction)
+      this.initialPageLoadInteraction = null
+      // Report metric on the initial page load time
+      handle(SUPPORTABILITY_METRIC_CHANNEL, ['SoftNav/Interaction/InitialPageLoad/Duration/Ms', Math.round(loadEventTime)], undefined, FEATURE_NAMES.metrics, this.ee)
+    })
+
+    this.latestRouteSetByApi = null
+    this.interactionInProgress = null // aside from the "page load" interaction, there can only ever be 1 ongoing at a time
+
+    this.blocked = false
+    this.waitForFlags(['spa']).then(([spaOn]) => {
```

Comment:
Doesnt have to happen as part of this initial PR but we should consider if we can gate all the event generation flows from happening if the spa flag is falsy

Reply:
You mean don't buffer stuff if spa is 0? We can. Make the instrument listeners check our abort signal, and pass that fn to aggregate to abort() here. Would be easier if we had a "parent" reference back to the instrument file, but can just pass it in like how domObserver is passed in, for now.
Do you want to proceed with this?

Reply:
yes but lets separate that work to a new ticket item, ty

Reply:
done, NR-217495